Mathlib/Topology/UniformSpace/Basic.lean

```diff
@@ -1596,12 +1598,33 @@ theorem mem_uniformity_of_uniformContinuous_invariant [UniformSpace α] [Uniform
   exact ⟨u, hu, fun a b c hab => @huvt ((_, _), (_, _)) ⟨hab, refl_mem_uniformity hv⟩⟩
 #align mem_uniformity_of_uniform_continuous_invariant mem_uniformity_of_uniformContinuous_invariant
 
+/-- An entourage of the diagonal in α and an entourage in β yield an entourage in α × β once we
+permute coordinates.-/
+def UniformityProd (u : Set (α × α)) (v : Set (β × β)) : Set ((α × β) × α × β) :=
+    { ((a₁, b₁), (a₂, b₂)) | (a₁, a₂) ∈ u ∧ (b₁, b₂) ∈ v }
```

Comment:
Could consider `Equiv.prodProdProdComm _ _ _ _ ⁻¹' (u ×ˢ v)`, though maybe it's not better.

Reply:
I've tried it, but in the end, I didn't see how it helped, and it made the definition slightly harder to parse for a human. The main issue being that there is almost no lemmas relating to `Equiv.prodProdProdComm`, and the little there is is easily `simp`-able, so I couldn't leverage this expression to make it useful.